kibble/datastore/file_system_datasource.go

```diff
@@ -0,0 +1,93 @@
+//    Copyright 2018 SHIFT72
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package datastore
+
+import (
+	"fmt"
+	"kibble/models"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/CloudyKit/jet"
+)
+
+var fileSystemArgs = []models.RouteArgument{}
+
+// FileSystemDataSource - represents a directory or .jet templates
+type FileSystemDataSource struct {
+}
+
+// GetName - name of the datasource
+func (ds *FileSystemDataSource) GetName() string {
+	return "FileSystem"
+}
+
+// GetRouteArguments returns the available route arguments
+func (ds *FileSystemDataSource) GetRouteArguments() []models.RouteArgument {
+	return fileSystemArgs
+}
+
+// GetEntityType - Get the entity type
+func (ds *FileSystemDataSource) GetEntityType() reflect.Type {
+	return reflect.TypeOf("")
+}
+
+// Iterator - loop over each film
+func (ds *FileSystemDataSource) Iterator(ctx models.RenderContext, renderer models.Renderer) (errCount int) {
+
+	// special case - only render the implicit root route if this is the default language context
+	// these files will not be rendered for every language
```

Comment:
If this does what the comment says then any site with multiple languages is not going to have different versions of any jet file that isn't a template right?

For example:
- https://www.kustendorf.online/en/signin.html
- https://www.kustendorf.online/signin.html

Reply:
Correct - just a small amount of magic. the "root" named route is a special case.

Reply:
This is how it currently works, /en/signin.html might be actually what we want though @PeculiarGoat

Reply:
Yeah, we do want those (sign in, register, forgot password, etc) to be available in all languages.

Reply:
@stajs @gscragg @PeculiarGoat 
Okay so this is not the current behaviour, but this is a good time to make this change.

I propose that by default we will render the directories of jet templates to all languages. You can opt out of this by naming the route "defaultLangOnly". A good example being the `.well-known` directory. Or shall we do something that is more language aware?

Reply:
Okay in discussion with @gscragg I have added `defaultLanguageOnly` flag to the route